### ISSUE TITLE
`reg_executing` is not cleared in two cases

### ISSUE THREAD

**Issue by zeertzjq** [vim/vim#10110]
### Steps to reproduce

Case 1:
1. Run `vim --clean`
2. Source the following file:
```vim
nnoremap s <Nop>
let @a = 's'
call feedkeys("@aqaq\<Esc>", 't')
```
3. A `q` is added to the buffer and the content of `@a` is not empty.

Case 2:
1. Run `vim --clean`
2. Source the following file:
```vim
call setline(1, 'aaa')
nnoremap s qa
let @a = 'fa'
call feedkeys("@asq\<Esc>", 't')
```
3. A `q` is added to the buffer and the content of `@a` is not empty.

Not sure if these can really be considered bugs, as the macro in case 1 only seems producible using `:let` (the keys in case 1 can be typed interactively though), and case 2 seems only reproducible when using `feedkeys()` with `t` flag. (Maybe it can be reproduced by typing these keys very quickly?)

### Expected behaviour

No `q` is added to the buffer and the content of `@a` becomes empty.

### Version of Vim

8.2.4651

### Environment

Operating system: Arch Linux
Terminal: kitty
Value of $TERM: xterm-kitty
Shell: fish

**Comment by zeertzjq** [comment]
Case 1 may be fixes by adding a check for clearing `reg_executing` in the loop in `vgetorpeek()`. Not sure if this is the best approach. Case 2 is caused by `vpeekc()` applying the mapping after the macro, causing chars from the mapping to be treated as a part of the macro. This can be hard to fix without breaking other things (e.g. autocommands that use the value of `reg_executing()`, or invoking `:normal` inside a macro).

**Comment by zeertzjq** [comment]
I'm trying to fix this in #10111.

**Comment by brammool** [comment]
Is there a better problem description for this issue?
